client/src/main/scala/com/dwolla/cloudflare/domain/model/accesscontrolrules/package.scala

```diff
@@ -0,0 +1,50 @@
+package com.dwolla.cloudflare.domain.model
+
+import com.dwolla.circe._
+import io.circe.generic.semiauto
+import io.circe._
+import shapeless.tag.@@
+
+package object accesscontrolrules {
+
+  type AccessControlRuleId = String @@ AccessControlRuleIdTag
+
+  private[cloudflare] val tagAccessControlRuleId: String => AccessControlRuleId = shapeless.tag[AccessControlRuleIdTag][String]
+
+}
+
+package accesscontrolrules {
+
+  import java.time.Instant
+
+  trait AccessControlRuleIdTag
+
+  case class AccessControlRule(id: Option[AccessControlRuleId] = None,
+                                  notes: Option[String],
+                                  allowed_modes: List[String] = List(),
+                                  mode: String,
+                                  configuration: AccessControlRuleConfiguration,
+                                  created_on: Option[Instant] = None,
+                                  modified_on: Option[Instant] = None,
+                                  scope: Option[AccessControlRuleScope] = None)
+
+  object AccessControlRule {
+    implicit val accessControlRuleCodec: Codec[AccessControlRule] = semiauto.deriveCodec
+  }
+
+  case class AccessControlRuleConfiguration(target: String,
```

Comment:
Could these strings be tagged to be more specific? It'd be worth going through all of the strings in this file asking that question.

Reply:
Addressed.

Just for future reference. What's the general rule of thumb for a whether string should be tagged? Is it along the lines of determining when a type would provide extra information that's useful within the domain? So, for example, for `AccessControlRule` I'd tag the type for the `id` field, but probably leave `notes` as a string because adding additional typing here doesn't really expose any useful information not already conveyed through the existing type.